drivers/tapdisk-nbdserver.c

```diff
@@ -283,17 +283,19 @@ tapdisk_nbdserver_set_free_request(td_nbdserver_client_t *client,
 
 void
 tapdisk_nbdserver_free_request(td_nbdserver_client_t *client,
-		td_nbdserver_req_t *req)
+			       td_nbdserver_req_t *req, bool free_client_if_dead)
 {
 	tapdisk_nbdserver_set_free_request(client, req);
-	if (unlikely(client->dead && !tapdisk_nbdserver_reqs_pending(client)))
-		tapdisk_nbdserver_free_client(client);
 
 	if (unlikely(client->n_reqs_free == (client->n_reqs / 4))) {
 		/* free requests, unmask the events */
 		tapdisk_server_mask_event(client->client_event_id, 0);
 	}
 
+	if (unlikely(free_client_if_dead &&
```

Comment:
I wonder if it would be better moving this into a separate function that callers can call if they need to rather than having an awkward sounding boolean variable?

Reply:
It's called, either directly or indirectly, from quite a number of places and whilst the the boolean arg is a bit icky and smelly I couldn't see making this something the callers have to do being any nicer.